File: PrismaFastApi/src/dtos/issue_dtos.py
from datetime import datetime
from typing import Optional
from src.dtos.decision_dtos import (
    DecisionOutgoingDto,
)
from src.dtos.discrete_probability_dtos import (
    DiscreteProbabilityOutgoingDto,
)
from src.dtos.outcome_dtos import (
    UtilityDto,
)
from src.dtos.uncertainty_dtos import (
    UncertaintyOutgoingDto,
)
from src.dtos.utility_dtos import (
    UtilityDto
)
from src.dtos.node_dtos import (
    NodeViaIssueOutgoingDto,
)
from src.dtos.shared_issue_node_dtos import IssueDto


class IssueOutgoingDto(IssueDto):
    type: str
    boundary: str
    node: NodeViaIssueOutgoingDto
    decision: Optional[DecisionOutgoingDto]
    uncertainty: Optional[UncertaintyOutgoingDto]
    utility: Optional[UtilityDto]
    created_at: datetime
    updated_at: datetime

File: PrismaDotnetApi/PrismaApi.Domain/Dtos/IssueDtos.cs
using System;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

public class IssueDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    [JsonPropertyName("project_id")]
    public Guid ProjectId { get; set; }
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;
    [JsonPropertyName("description")]
    public string Description { get; set; } = string.Empty;
    [JsonPropertyName("order")]
    public int Order { get; set; }
}

public class IssueIncomingDto : IssueDto
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "Unassigned";
    [JsonPropertyName("boundary")]
    public string Boundary { get; set; } = "out";
    [JsonPropertyName("node")]
    public NodeIncomingDto? Node { get; set; }
    [JsonPropertyName("decision")]
    public DecisionIncomingDto? Decision { get; set; }
    [JsonPropertyName("uncertainty")]
    public UncertaintyIncomingDto? Uncertainty { get; set; }
    [JsonPropertyName("utility")]
    public UtilityIncomingDto? Utility { get; set; }
}

public class IssueOutgoingDto : IssueDto
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "Unassigned";
    [JsonPropertyName("boundary")]
    public string Boundary { get; set; } = "out";
    [JsonPropertyName("node")]
    public NodeViaIssueOutgoingDto Node { get; set; } = new();
    [JsonPropertyName("decision")]
    public DecisionOutgoingDto? Decision { get; set; }
    [JsonPropertyName("uncertainty")]
    public UncertaintyOutgoingDto? Uncertainty { get; set; }
    [JsonPropertyName("utility")]
    public UtilityOutgoingDto? Utility { get; set; }
    [JsonPropertyName("created_at")]
    public DateTimeOffset CreatedAt { get; set; }
    [JsonPropertyName("updated_at")]
    public DateTimeOffset UpdatedAt { get; set; }
}

public class IssueViaNodeOutgoingDto : IssueDto
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "Unassigned";
    [JsonPropertyName("boundary")]
    public string Boundary { get; set; } = "out";
    [JsonPropertyName("decision")]
    public DecisionOutgoingDto? Decision { get; set; }
    [JsonPropertyName("uncertainty")]
    public UncertaintyOutgoingDto? Uncertainty { get; set; }
    [JsonPropertyName("utility")]
    public UtilityOutgoingDto? Utility { get; set; }
}
